### ISSUE TITLE
how do I assign new values to the gradients of embeddings?

### ISSUE THREAD

**Issue by gray311** [Luodian/Otter#117]
Hello, after using DeepSpeed, how do I assign new values to the gradients of embeddings? I have tried some methods, but they do not seem to be very effective.
![屏幕截图 2023-05-25 105706](https://github.com/Luodian/Otter/assets/64787301/785e19bc-496a-43df-bf7c-4e1f3d9607ce)

**Comment by Luodian** [comment]
It's very tricky to reset the params of token embeddings. The `mask_gradients` method in traditional training script didnt work for deepspeed since you cant get tensor's gradient during training in ds framework (https://github.com/microsoft/DeepSpeed/issues/3310). According to our new experiment, you can directly train all

**Comment by gray311** [comment]
ook, thinks for your reply
